Zotero/Controllers/SpeechManager.swift

```diff
@@ -0,0 +1,402 @@
+//
+//  SpeechManager.swift
+//  Zotero
+//
+//  Created by Michal Rentka on 11.03.2025.
+//  Copyright © 2025 Corporation for Digital Scholarship. All rights reserved.
+//
+
+import AVFoundation
+import NaturalLanguage
+
+import CocoaLumberjackSwift
+import RxCocoa
+import RxSwift
+
+protocol SpeechmanagerDelegate: AnyObject {
+    associatedtype Index: Hashable
+
+    func getCurrentPageIndex() -> Index
+    func getNextPageIndex(from currentPageIndex: Index) -> Index?
+    func getPreviousPageIndex(from currentPageIndex: Index) -> Index?
+    func text(for indices: [Index], completion: @escaping ([Index: String]?) -> Void)
+    func moved(to pageIndex: Index)
+}
+
+final class SpeechManager<Delegate: SpeechmanagerDelegate>: NSObject, AVSpeechSynthesizerDelegate {
+    enum State {
+        case speaking, paused, stopped, loading
+
+        var isStopped: Bool {
+            switch self {
+            case .speaking, .loading, .paused:
+                return false
+
+            case .stopped:
+                return true
+            }
+        }
+    }
+
+    private enum Error: Swift.Error {
+        case cantGetText
+    }
+
+    private struct SpeechData {
+        let startIndex: Int
+        let speakingRange: NSRange
+
+        func copy(with localRange: NSRange) -> SpeechData {
+            return SpeechData(startIndex: startIndex, speakingRange: localRange)
+        }
+
+        var globalRange: NSRange {
+            return NSRange(location: startIndex + speakingRange.location, length: speakingRange.length)
+        }
+    }
+
+    private struct PageData {
+        // Voice used for this page
+        let voice: AVSpeechSynthesisVoice
+        // Text to read
+        let text: String
+
+        init(text: String, voice: AVSpeechSynthesisVoice) {
+            self.voice = voice
+            self.text = text
+        }
```

Comment:
Can be simplified as such:
```suggestion
        // Text to read
        let text: String
        // Voice used for this page
        let voice: AVSpeechSynthesisVoice
```